src/wallet/rpc/backup.cpp

```diff
@@ -93,6 +93,20 @@ static void RescanWallet(CWallet& wallet, const WalletRescanReserver& reserver,
     }
 }
 
+static void EnsureBlockDataFromTime(interfaces::Chain& chain, int64_t timestamp)
+{
+    if (!chain.havePruned())
+        return;
+
+    int height{0};
+    const bool found{chain.findFirstBlockWithTimeAndHeight(timestamp - TIMESTAMP_WINDOW, 0, FoundBlock().height(height))};
+    const std::optional<int> chainHeight{chain.getHeight()};
+
+    if (found && chainHeight && !chain.hasBlocks(chain.getBlockHash(chainHeight.value()), height)) {
+        throw JSONRPCError(RPC_WALLET_ERROR, "Pruned blocks required to import keys");
```

Comment:
I think that at this point we have the required `height` for which we need blocks. Could this change to a more descriptive error message stating that required weight? And maybe "[...]. Use RPC call getblockchaininfo to determine your pruned height" as [here](https://github.com/bitcoin/bitcoin/blob/15692e2641592394bdd4da0a7c2d371de8e576dd/src/wallet/rpc/transactions.cpp#L898)?

Reply:
thanks, added it